pkg/controller/endpointslice/topologycache/topologycache.go

```diff
@@ -35,11 +37,12 @@ const (
 
 // TopologyCache tracks the distribution of Nodes and endpoints across zones.
 type TopologyCache struct {
-	lock               sync.Mutex
-	sufficientNodeInfo bool
-	cpuByZone          map[string]*resource.Quantity
-	cpuRatiosByZone    map[string]float64
-	endpointsByService map[string]map[discovery.AddressType]EndpointZoneInfo
+	lock                    sync.Mutex
+	sufficientNodeInfo      bool
+	cpuByZone               map[string]*resource.Quantity
+	cpuRatiosByZone         map[string]float64
+	endpointsByService      map[string]map[discovery.AddressType]EndpointZoneInfo
+	hintsPopulatedByService map[string]bool
```

Comment:
As I'm thinking about this more, it might be _slightly_ cleaner to use the internal k/k sets.String helper here to just track the `namespace/name` of services that have hints populated.

Reply:
not sure I get it

Reply:
So my rough idea is that it's not useful to track Services that don't have hints populated, we can just assume that's the default state. Instead, we can use the kubernetes `sets.String` util to get set-like behavior here to quickly understand if a Service has hints populated. It's nearly the same thing, it's just a `map[string]struct{}`, but it has a slightly nicer interface to work with + would mean we'd only store Services that had hints populated.

Reply:
on both cases we stored only Services that had hints populated, no?

Changed to a Set though, PTAL @robscott

Reply:
I'm not sure, map[string]bool seems to indicate that the value could be false, but maybe it was just always true when present?

Reply:
could be false but we never set the value to false. Anyway, I agree that s Set is more elegant and efficient here.